### PR TITLE
WL-3675 Checks if the email address already exists . If the userId is different from email address

### PR DESCRIPTION
**Pull request by surya-janani**
it sends out validation link.

### CODE REVIEW

**Comment by surya-janani** [comment]
Calling overloaded method

File: user-tool/tool/src/java/org/sakaiproject/user/tool/UsersAction.java [hunk 1]
```diff
@@ -801,7 +809,12 @@ public void doSave(RunData data, Context context)
 		{
 			try
 			{
-				UserDirectoryService.commitEdit(edit);
+				String valueEmail = (String)state.getAttribute("valueEmail");
+				if(EmailValidator.getInstance().isValid(edit.getEid())&& !edit.getEid().equalsIgnoreCase(valueEmail)) {
+					validationLogic.createValidationAccount(edit.getId(), ValidationAccount.ACCOUNT_STATUS_USERID_UPDATE,valueEmail);
+					addAlert(state,"A validation link has been sent to  "+valueEmail+" Please validate to update your UserId");
+				}
+					UserDirectoryService.commitEdit(edit);
```
Comment (buckett) [inline user-tool/tool/src/java/org/sakaiproject/user/tool/UsersAction.java]: Indentation.

File: user-tool/tool/src/java/org/sakaiproject/user/tool/UsersAction.java [hunk 2]
```diff
@@ -801,7 +809,12 @@ public void doSave(RunData data, Context context)
 		{
 			try
 			{
-				UserDirectoryService.commitEdit(edit);
+				String valueEmail = (String)state.getAttribute("valueEmail");
+				if(EmailValidator.getInstance().isValid(edit.getEid())&& !edit.getEid().equalsIgnoreCase(valueEmail)) {
```
Comment (buckett) [inline user-tool/tool/src/java/org/sakaiproject/user/tool/UsersAction.java]: Formatting doesn't match surrounding code.

File: user-tool/tool/src/java/org/sakaiproject/user/tool/UsersAction.java [hunk 3]
```diff
@@ -801,7 +809,12 @@ public void doSave(RunData data, Context context)
...
+					validationLogic.createValidationAccount(edit.getId(), ValidationAccount.ACCOUNT_STATUS_USERID_UPDATE,valueEmail);
+					addAlert(state,"A validation link has been sent to  "+valueEmail+" Please validate to update your UserId");
```
Comment (buckett) [inline user-tool/tool/src/java/org/sakaiproject/user/tool/UsersAction.java]: Needs internationalizations.

File: user-tool/tool/src/java/org/sakaiproject/user/tool/UsersAction.java [hunk 4]
```diff
@@ -102,6 +105,11 @@
 	private static final String IMPORT_EMAIL="email";
 	private static final String IMPORT_PASSWORD="password";
 	private static final String IMPORT_TYPE="type";
+	ValidationLogic validationLogic;
```
Comment (buckett) [inline user-tool/tool/src/java/org/sakaiproject/user/tool/UsersAction.java]: Why have this package visibility and not private?

File: user-tool/tool/src/java/org/sakaiproject/user/tool/UsersAction.java [hunk 5]
```diff
@@ -1096,9 +1109,20 @@ else if ((mode != null) && (mode.equalsIgnoreCase("edit")) && (!singleUser))
...
 				return false;
 		}
-		
+
 		// get the user
 		UserEdit user = (UserEdit) state.getAttribute("user");
+		if(email != null & !(email.equalsIgnoreCase(user.getEid()))) {
+			try {
+				User userExist = UserDirectoryService.getUserByEid(email);
+				if(userExist != null) {
```
Comment (buckett) [inline user-tool/tool/src/java/org/sakaiproject/user/tool/UsersAction.java]: It never returns null so you don't need to check it.

**Comment by surya-janani** [comment]
Made review changes. Yet to rebase

**Comment by buckett** [comment]
This doesn't merge,

**Pull request merged**